packages/api/src/challenges/challengeProvider.ts

```diff
@@ -0,0 +1,20 @@
+export type Challenge = {
+  challengeID: string
+  challenge: string
+  validUntil: string
+}
+export type ChallengeValidationProps = {
+  challengeID: string
+  solve: number
+}
+
+export type ChallengeValidationReturn = {
+  result: string
+  msg: string
+  valid: boolean
```

Comment:
msg or new message is human readable message while result is a string to react on a specific reason why it failed. We don't use this for now but it can be interesting in the future. I can remove this but i guess at some point we will be happy to have this.